### PR TITLE
Refactor OfflineStoreConfig classes into their owning modules

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
Follow up from #1649. This refactors the OfflineStoreConfig in the same exact way so that things are loaded dynamically instead of hardcoding in the `repo_config.py` module.
 
**Which issue(s) this PR fixes**:

Part 2 for #1605

**Does this PR introduce a user-facing change?**:

```release-note
Allow OfflineStore implementations to be loaded dynamically.
```

### CODE REVIEW

File: sdk/python/feast/errors.py [hunk 1]
```diff
@@ -85,10 +85,11 @@ def __init__(self, online_store_class_name: str):
         )
 
 
-class FeastOnlineStoreConfigInvalidName(Exception):
-    def __init__(self, online_store_config_class_name: str):
+class FeastStoreConfigInvalidName(Exception):
+    def __init__(self, online_store_config_class_name: str, store_type="Online"):
```
Comment (woop) [inline sdk/python/feast/errors.py:89]: Is there a reason to set a default here?

Comment (achals) [reply sdk/python/feast/errors.py:89]: Nope, going to clean this up

Comment (woop) [reply sdk/python/feast/errors.py:89]: Also, should `online_store_config_class_name` be online store agnostic?

File: sdk/python/feast/errors.py [hunk 2]
```diff
@@ -85,10 +85,11 @@ def __init__(self, online_store_class_name: str):
...
         super().__init__(
-            f"Online Store Config Class '{online_store_config_class_name}' should end with the string `OnlineStoreConfig`.'"
+            f"Online Store Config Class '{online_store_config_class_name}' "
+            f"should end with the string `{store_type}Config`.'"
```
Comment (woop) [inline sdk/python/feast/errors.py:92]: It's not clear what `OnlineConfig` here means. Shouldn't this be the name of the Config class? (assuming the default value is used)

File: sdk/python/feast/errors.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/errors.py]: done

File: sdk/python/feast/errors.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/errors.py:92]: Sorry, I don't know what you're referring to here - the diff may have changed since the last time you looked at it

File: sdk/python/feast/infra/online_stores/sqlite.py [hunk 3]
```diff
@@ -34,7 +34,9 @@
 class SqliteOnlineStoreConfig(FeastConfigBaseModel):
     """ Online store config for local (SQLite-based) store """
 
-    type: Literal["sqlite"] = "sqlite"
+    type: Literal[
+        "sqlite", "feast.infra.online_stores.sqlite.SqliteOnlineStore"
```
Comment (tsotnet) [inline sdk/python/feast/infra/online_stores/sqlite.py:38]: why is this change necessary?

File: sdk/python/feast/infra/provider.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def get_provider(config: RepoConfig, repo_path: Path) -> Provider:
             # The original exception can be anything - either module not found,
             # or any other kind of error happening during the module import time.
             # So we should include the original error as well in the stack trace.
-            raise errors.FeastModuleImportError(module_name) from e
+            raise errors.FeastModuleImportError(module_name, "provider") from e
```
Comment (tsotnet) [inline sdk/python/feast/infra/provider.py:166]: nit: capitalize provider

File: sdk/python/feast/repo_config.py [hunk 5]
```diff
@@ -235,11 +207,33 @@ def get_online_config_from_type(online_store_type: str):
...
+
+def get_offline_config_from_type(offline_store_type: str):
+    if offline_store_type in OFFLINE_STORE_CLASS_FOR_TYPE:
+        offline_store_type = OFFLINE_STORE_CLASS_FOR_TYPE[offline_store_type]
+    else:
+        assert offline_store_type.endswith("OfflineStore")
+    module_name, offline_store_class_type = offline_store_type.rsplit(".", 1)
+    config_class_name = f"{offline_store_class_type}Config"
+
+    return get_config_class_from_type(module_name, config_class_name, "OfflineStore")
```
Comment (tsotnet) [inline sdk/python/feast/repo_config.py:234]: These 2 can also be deduplicated. For example, you could construct
```
STORE_CLASS_FOR_TYPE = {
    "OnlineStore": ONLINE_STORE_CLASS_FOR_TYPE,
    "OfflineStore": OFFLINE_STORE_CLASS_FOR_TYPE,
}

def get_config_class_from_type(store_type: str, store_class_type: str):
    if store_class_type in STORE_CLASS_FOR_TYPE[store_type]:
        store_class_type = STORE_CLASS_FOR_TYPE[store_type][store_class_type]
    else:
        assert store_class_type.endswith(store_type)
    module_name, store_class_type = store_class_type.rsplit(".", 1)
    config_class_name = f"{store_class_type}Config"

    # merge existing get_config_class_from_type function (line 188) here instead of splitting
    return get_config_class_from_type(module_name, config_class_name, store_type)
```

Ideally, we shouldn't have Providers handled in a different way, but it's a little bit different (e.g. no Config classes there), so I'll leave it up to you if you think we can deduplicate that in a reasonable way.

**Review by tsotnet — CHANGES_REQUESTED** [review]

File: sdk/python/feast/infra/provider.py (hunk 4 above)
Comment (achals) [reply sdk/python/feast/infra/provider.py:166]: done

File: sdk/python/feast/repo_config.py (hunk 5 above)
Comment (achals) [reply sdk/python/feast/repo_config.py]: I pulled in the provider to use a shared code path for loading.

I'm tempted to not introduce `STORE_CLASS_FOR_TYPE` cause the additional indirection/method doesn't seem that bad, and seems more legible. WDYT?

Comment (achals) [reply sdk/python/feast/repo_config.py]: Take a look at the latest version, it's more cleaned up

File: sdk/python/feast/infra/online_stores/sqlite.py (hunk 3 above)
Comment (tsotnet) [reply sdk/python/feast/infra/online_stores/sqlite.py:38]: Gotcha, makes sense.

**Review by tsotnet — APPROVED** [review]
